libs/MeshKernel/include/MeshKernel/Definitions.hpp

```diff
@@ -86,6 +86,16 @@ namespace meshkernel
         N  ///< N-direction
     };
 
+    /// @enum InterpolationValuesTypes
+    /// @brief The possible types of the values to be interpolated in the gridded sample 
+    enum class InterpolationValuesTypes
+    {
+        shortType = 0, ///< short type
+        intType = 1,   ///< int type
+        floatType = 2, ///< float type
+        doubleType = 3 ///< double type
```

Comment:
Just a comment.
Since the interpolation is going to be restricted to these four scalar types, could move all the implementation detail to the implementation file and also add


`template class meshkernel::BilinearInterpolationOnGriddedSamples<float>;`
`template class meshkernel::BilinearInterpolationOnGriddedSamples<double>;`
`template class meshkernel::BilinearInterpolationOnGriddedSamples<int>;`
`template class meshkernel::BilinearInterpolationOnGriddedSamples<short>;`

This may help to reduce compilation times, if this proves necessary